lib/openstack_api/openstack_service.py

```diff
@@ -0,0 +1,50 @@
+from typing import Optional
+from openstack.connection import Connection
+from openstack.compute.v2.service import Service
+
+
+def disable_service(
+    conn: Connection,
+    service: Service,
+    hypervisor_name: str,
+    service_binary: str,
+    disabled_reason: str,
+) -> Optional[Service]:
+    """
+    Disables an Openstack service
+    :param conn: Openstack connection
+    :param service: The instance of the service class.
+    :param hypervisor_name: (str): The name or ID of the hypervisor.
+    :param service_binary: The name of the service.
+    :param disabled_reason: The reason for disabling the service.
+    """
+
+    if service.status == "enabled":
+        return conn.compute.disable_service(
+            service,
+            host=hypervisor_name,
+            binary=service_binary,
+            disabled_reason=disabled_reason,
+        )
+    return None
```

Comment:
Food for thought:

I wonder if an error should be raised rather than returning `None`.
E.g.
```suggestion
    raise RunetimeError(f"Failed to disable OpenStack service {binary} on host {host}. Reason: Service is already disabled.")
```

Reply:
I had something very similar in before, but it was running on a `print` statement rather than a `RuntimeError`. After a discussion with Anish and for the simplicity of the unit tests, we went ahead with returning `None` instead.

Reply:
That is a very valid scenario and I'm sort of on the fence about this now because as I mentioned I did have an error printing earlier but Anish made a compelling argument to not have to have one. You both have a compelling argument really. 

EDIT: After playing around with StackStorm and looking at the logs after disabling and enabling an HV, I think I would agree, I would like a `RuntimeError` or something similar to print out.

We can search by status using a seperate stackstorm action to check if the HV is enabled/disabled, but it does take extra steps.

Happy to show you how it all works in stackstorm and perhaps we can go over the code together.

Reply:
Sounds good